packages/eas-cli/src/credentials/ios/IosCredentialsProvider.ts

```diff
@@ -143,17 +143,32 @@ export default class IosCredentialsProvider {
 
   private assertProvisioningProfileType(provisioningProfile: string, targetName?: string): void {
     const isAdHoc = isAdHocProfile(provisioningProfile);
-    if (this.options.distribution === 'internal' && !isAdHoc) {
-      throw new Error(
-        `You must use an adhoc provisioning profile${
-          targetName ? ` (target '${targetName})'` : ''
-        } for internal distribution`
-      );
-    } else if (this.options.distribution !== 'internal' && isAdHoc) {
+    const isEnterprise = isEnterpriseUniversalProfile(provisioningProfile);
+    if (this.options.distribution === 'internal') {
+      if (this.options.enterpriseProvisioning === 'universal' && !isEnterprise) {
+        throw new Error(
+          `You must use a universal provisioning profile${
+            targetName ? ` (target '${targetName})'` : ''
+          } for internal distribution if you specified "enterpriseProvisioning": "universal" in eas.json`
+        );
+      } else if (this.options.enterpriseProvisioning === 'adhoc' && !isAdHoc) {
+        throw new Error(
+          `You must use an adhoc provisioning profile${
+            targetName ? ` (target '${targetName})'` : ''
+          } for internal distribution if you specified "enterpriseProvisioning": "adhoc" in eas.json`
+        );
+      } else if (!this.options.enterpriseProvisioning && !isEnterprise && !isAdHoc) {
```

Comment:
As far as I understand not setting `enterpriseProvisioning` implicitly defaults to `adhoc`.

> [iOS](https://docs.expo.dev/build/internal-distribution/#ios)
> The configuration above tells EAS Build that you would like to use ad hoc distribution, which is available for all paid Apple Developer accounts. It is not available for free accounts.

So I think you can combine those two else if's to just:
```
     else if (!isAdHoc) {
        throw new Error(
          `You must use an adhoc provisioning profile${
            targetName ? ` (target '${targetName})'` : ''
          } for internal distribution if you specified "enterpriseProvisioning": "adhoc" in eas.json`
        );
      }
```

Reply:
There is no default, For managed credentials, If you are on regular account only AdHoc is available, so we are picking that automatically, but for Enterprise accounts you will be prompted whether you want to use AdHoc or universal provisioning. For local credentials there is no need for a prompt, because we never generate new profiles, so we don't need declaration what type it should be.